gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/DiskStoreImpl.java

```diff
@@ -5268,4 +5277,11 @@ else if (memoryState.isEviction()) {
       resetAsyncQueueCapacity();
     }
   }
+
+  public void clearRecoveryMapOfAllOplogs() {
+    PersistentOplogSet oplogSet = getPersistentOplogSet(null);
+    for (Oplog oplog : oplogSet.getAllOplogs()) {
+      oplog.clearInitRecoveryMap();
+    }
+  }
 }
```

Comment:
Shouldn't these index/oplog related changes (recreate indexes I suppose) be part of a separate PR? This is a different issue that should have a separate JIRA associated with it, no?